frontend/public/components/nav/nav-header.tsx

```diff
@@ -1,37 +1,117 @@
 import * as React from 'react';
 import { connect } from 'react-redux';
-import { Title } from '@patternfly/react-core';
-import { getActivePerspective } from '../../reducers/ui';
+import {Dropdown, DropdownItem, DropdownToggle, Title} from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons';
 import * as plugins from '../../plugins';
-import { RootState } from '../../redux';
+import {RootState} from '../../redux';
+import { getActivePerspective } from '../../reducers/ui';
+import * as UIActions from '../../actions/ui';
+import { history } from '../utils';
 
 type StateProps = {
   activePerspective: string;
 };
 
-export const NavHeader: React.FC<StateProps> = ({ activePerspective }) => {
-  const { icon, name } = plugins.registry
-    .getPerspectives()
-    .find((p) => p.properties.id === activePerspective).properties;
-
-  return (
-    <div className="oc-nav-header">
-      <Title className="oc-nav-header__title" size="md">
-        {icon} {name}
-      </Title>
-    </div>
-  );
+export type NavHeaderProps = {
+  activePerspective?: string;
+  setActivePerspective?: any;
+  onPerspectiveSelected: () => void;
+};
+
+export type NavHeaderState = {
+  isPerspectiveDropdownOpen: boolean;
 };
 
+class NavHeader_ extends React.Component<NavHeaderProps, NavHeaderState> {
```

Comment:
nit:
```suggestion
class NavHeader extends React.Component<NavHeaderProps, NavHeaderState> {
```

Console is moving towards more functional components and hooks. The class component looks ok though.